### ISSUE TITLE
Focal loss would lead to nan?

### ISSUE THREAD

**Issue by liming-ai** [kaidic/LDAM-DRW#17]
Hi @kaidic 

Thanks for your fantastic work, but when I tried to reproduce the focal loss result, I found that when gamma=0.5, the focal loss would lead to nan loss during training, but the focal loss in [this repo](https://github.com/clcarwin/focal_loss_pytorch/blob/master/focalloss.py) can make it.

I checked the two different designed focal loss carefully and found the forward progress of them are the same but model parameters became different after backward, I am quite confused, could you please give me some advice?

Thanks for your contribution again!

**Comment by liming-ai** [comment]
The output loss calculated by `F.cross_entropy_loss` could be negative, lead to nan loss. We can add a small value or use `F.relu()` to deal with this issue. This problem showed in my device with `PyTorch==1.8.0`, hope it is helpful

![image](https://user-images.githubusercontent.com/73068772/128675720-560b5bb2-5586-47aa-b90a-6edbecf2c36b.png)
